inject.py

```diff
@@ -0,0 +1,148 @@
+try:
+    import ujson as json
+except ImportError:
+    import json  # noqa: F401
+
+from auth import connect
+
+
+opts = {
+    'index': 'gatest',
+    'type': '_doc',
+    'with_retry': True,
+    'timeout': 30,
+    'bulk_size': 400,
+    'progress': False,
+    'update': False,
+    'delete': False,
+    'id_field': False,
+    'as_child': False,
+    'verify_certs': False,  
+}
+
+
+def grouper(iterable, n, fillvalue=None):
+    """Collect data into fixed-length chunks or blocks"""
+    args = [iter(iterable)] * n
+    return zip_longest(fillvalue=fillvalue, *args)
+
+
+def bulk_builder(bulk, config):
+    """Build bulk insert entries"""
+    for item in filter(None, bulk):
+        body = {
+            '_index': config['index'],
+            '_type': config['type'],
+            '_source': item
+        }               
+        if config['id_field']:
+            body['_id'] = item[config['id_field']]
+            if config['as_child']:
+                body['_parent'] = body['_id']
+                body['_routing'] = body['_id']
+        if config['update']:
+            # default _op_type is 'index', which will overwrites existing doc
+            body['_op_type'] = 'update' 
+            body['doc'] = item
+            del body['_source']           
+        yield body
+
+
+def json_lines_iter(fle):
+    """Yield json objects from a file"""
+    for line in fle:
+        if isinstance(line, str):
+            yield json.loads(line)
+        else:
+            yield json.loads(line.decode('utf-8'))
+        
+
+def single_bulk_to_es(bulk, config, attempt_retry, es):
+    bulk = bulk_builder(bulk, config)
+    max_attempt = 1
+    if attempt_retry:
+        max_attempt += 3
+    for attempt in range(1, max_attempt + 1):
+        print('Attempting chunk')
+        try:
+            z=helpers.bulk(es, bulk, chunk_size=config['bulk_size'])
+            print(z[0])
+        except Exception as e:
+            if attempt < max_attempt:
+                wait_seconds = attempt * 3
+                log('warn', 'attempt [%s/%s] got Xception, will retry after %s seconds\n%s' % (attempt, max_attempt, wait_seconds, e))
+                time.sleep(wait_seconds)
+                continue
+            log('error', 'attempt [%s/%s] Retrying failed, data not sent.' % (attempt, max_attempt))
+            raise e
+        if attempt > 1:
+            log('info', 'attempt [%s/%s] Success! Recovered from previous error.' % (attempt, max_attempt))
+        # completed succesfully
+        break
+
+
+def load(lines, config, es):
+    bulks = grouper(lines, config['bulk_size'] * 3)
+    for i, bulk in enumerate(bulks):
+        if i % 100 == 0:
+            print(f'Done {i} records!')
+        try:
+            single_bulk_to_es(bulk, config, config['with_retry'], es)
+        except Exception as e:
+            log('warn', 'Chunk {i} got exception ({e}) while processing'.format(e=e, i=i))
+
+
+def format_msg(msg, sevirity):
```

Comment:
Is this supposed to be `severity` ?

Reply:
Fixed. Actually went with 'level' as log levels are technically a mixture of priority/severity/hierarchy depending on context.